src/Microsoft.Crank.Agent/Startup.cs

```diff
@@ -4961,6 +4941,15 @@ private void OnShutdown()
         {
             try
             {
+                Log.WriteLine("Cancelling remaining jobs");
+
+                if (!_processJobsCts.IsCancellationRequested)
+                {
+                    _processJobsCts.Cancel();
+
+                    Task.WaitAny(_processJobsTask, Task.Delay(TimeSpan.FromSeconds(30)));
```

Comment:
Why is it blocking here?

Reply:
I want to let the currently running jobs, if any, to be stopped, or forced to stop, such that there is no ghost process holding on http ports. Without that the unit tests would fail, showing the issue.

Reply:
Or is the question about "here"?